### DISCUSSION TITLE
How to make changes to frontend without rebuilding the entire application and losing data?

### DISCUSSION THREAD

**Discussion by cptalpdeniz** [getredash/redash#6642] [Q&A] [answered]
I have downloaded version 10, made few changes to the source code and using the local development guide successfully deployed the application to my server. The question now is that though, if i want to make changes on the frontend without losing data and reinstall all the time (`make down` removes everything) how do I do that? I tried docker-compose down and few other commands as well but everytime it seems like the database is removed and I have to run `make create_database` which is a big problem.

Looking for suggestions. Thanks!

**Comment by ilsaloving** [comment, marked as answer]
This entirely depends on the contents of your docker-compose.yaml file.  If you need the DB to persist, then you have to either use a persistent volume, or bind /var/lib/postgresql/data to a local folder.

**Comment by cptalpdeniz** [comment]
Closing this since the project is cancelled.
